pycalphad/core/minimizer.pyx

```diff
@@ -750,13 +751,16 @@ cpdef find_solution(list compsets, int num_statevars, int num_components,
     step_size = 1./10
     for iteration in range(1000):
         state.iteration = iteration
-        if (state.mass_residual > 10) and (np.max(np.abs(state.chemical_potentials)) > 1.0e10):
+        if (state.mass_residual > 10) and (np.any(np.abs(state.chemical_potentials) > 1.0e10)):
             state.chemical_potentials[:] = spec.initial_chemical_potentials
 
-        old_state = copy(state)
+        previous_chemical_potentials[:] = state.chemical_potentials[:]
+        previous_phase_amt[:] = state.phase_amt[:]
+        previous_phase_compositions[:, :] = state.phase_compositions[:, :]
+
         take_step(spec, state, step_size)
-        delta_phase_amt = np.array(state.phase_amt) - np.array(old_state.phase_amt)
-        if ((state.mass_residual > 1e-2) and (not np.all(np.array(state.chempot_diff) < 1.0))) or (iteration == 0):
+        delta_phase_amt = np.asarray(state.phase_amt) - np.asarray(previous_phase_amt)
```

Comment:
Profiling would tell you if it makes a difference or not, but this could be rewritten to avoid allocating a new array if you use a `for` loop.

Reply:
Resolving because this code is removed